Packs/ApiModules/Scripts/OktaApiModule/OktaApiModule.py

```diff
@@ -0,0 +1,252 @@
+                missing_required_params.append('Scopes')
+
+            if not jwt_algorithm:
+                missing_required_params.append('JWT algorithm')
+
+            if not private_key:
+                missing_required_params.append('Private key')
+
+            if missing_required_params:
+                raise ValueError(f'Required OAuth parameters are missing: {", ".join(missing_required_params)}')
+
+            # Set type of variables non-optional after we assured they're assigned for mypy
+            self.client_id: str = client_id  # type: ignore
+            self.scopes: list[str] = scopes  # type: ignore
+            self.private_key: str = private_key  # type: ignore
+            self.jwt_algorithm: JWTAlgorithm = jwt_algorithm  # type: ignore
+
+        self.initial_setup()
+
+    def assign_app_role(self, client_id: str, role: str, auth_type: AuthType = AuthType.API_TOKEN) -> dict:
+        """
+        Assign a role to a client application.
+
+        Args:
+            client_id (str): The ID of the client application.
+            role (str): The role to assign to the client application.
+            auth_type (bool | None, optional): Authentication type to use for the request. Defaults to AuthType.API_TOKEN.
+
+        Returns:
+            dict: The response from the API.
+        """
+        return self._http_request(
+            auth_type=auth_type,
+            url_suffix=f'/oauth2/v1/clients/{client_id}/roles',
+            method='POST',
+            json_data={
+                'type': role,
+            },
+        )
+
+    def generate_jwt_token(self, url: str) -> str:
+        """
+        Generate a JWT token to use for OAuth authentication.
+
+        Args:
+            url (str): The URL to use for the JWT token (for the 'aud' claim).
+
+        Returns:
+            str: The JWT token.
+        """
+        current_time = datetime.utcnow()
+        expiration_time = current_time + timedelta(minutes=TOKEN_EXPIRATION_TIME)
+
+        return jwt.encode(
+            payload={
+                'aud': url,
+                'iat': int((current_time - datetime(1970, 1, 1)).total_seconds()),
+                'exp': int((expiration_time - datetime(1970, 1, 1)).total_seconds()),
+                'iss': self.client_id,
+                'sub': self.client_id,
+                'jti': str(uuid.uuid4()),
+            },
+            key=self.private_key,
+            algorithm=self.jwt_algorithm.value,
+        )
+
+    def generate_oauth_token(self, scopes: list[str]) -> dict:
+        """
+        Generate an OAuth token to use for authentication.
+
+        Args:
+            scopes (list[str]): A list of scopes to request for the token.
+
+        Returns:
+            dict: The response from the API.
+        """
+        auth_url = self._base_url + '/oauth2/v1/token'
+        jwt_token = self.generate_jwt_token(url=auth_url)
+
+        return self._http_request(
+            auth_type=AuthType.NO_AUTH,
```

Comment:
relevant, because of line 229 `auth_type = auth_type if auth_type is not None else self.auth_type` which is `AuthType.API_TOKEN` by default

Reply:
It's needed here since this specific call (generating the token) requires no authentication header (a private key is used to sign the JWT token we generate here, which is the actual authentication method used).
When the API call was made with an "Authorization" header set on my testings, the API call failed.

This request is the only reason `AuthType.NO_AUTH` exists (so that we omit the `Authorization` header).
You can see this logic on the `_http_request` I wrote for the module (the authorization header is not set for `AuthType.NO_AUTH` requests).

Reply:
after the changes:
now as you use `_http_request` the `auth_type` is not relevant and confusing